app/views/RoomInfoView/index.js

```diff
@@ -189,19 +197,25 @@ class RoomInfoView extends React.Component {
 					const { user } = result;
 					const { roles } = user;
 					if (roles && roles.length) {
-						user.parsedRoles = await Promise.all(
-							roles.map(async role => {
-								const description = await this.getRoleDescription(role);
-								return description;
-							})
-						);
+						user.parsedRoles = await this.parseRoles(roles);
 					}
 
 					this.setState({ roomUser: user });
 				}
 			} catch {
 				// do nothing
 			}
+		} else {
+			try {
+				const { roles } = roomUser;
+				if (roles && roles.length) {
+					roomUser.parsedRoles = await this.parseRoles(roles);
```

Comment:
As I understand it, are you here manipulating the state directly?
I don't think this is a good practice, it can generate several side-effects.
The right thing would be to create a variable receiving the parseRoles and then unstructure it to receive the value.

Reply:
> if (roles && roles.length) {
>   const parsedRoles = await this.parseRoles(roles);
>   this.setState({ roomUser: { ...roomUser, parsedRoles } });
> } else {
>   this.setState({ roomUser });
> }

Reply:
I committed the changes